### DISCUSSION TITLE
How to add UI from my own UI Library?

### DISCUSSION THREAD

**Discussion by martinrhan** [stride3d/stride#2407] [Q&A] [answered]
In editor of my UI Page, I can only add UI to visual tree from Standard UI Library. I don't know how to add from my own Library

**Comment by Basewq** [comment]
For your own UI library controls to appear, you need to name the root control(s).

eg. 
![image](https://github.com/user-attachments/assets/7956a162-4a33-45c4-bfc5-429ed342609c)
Either:
- Select the control in the Visual Tree, press `F2` then rename it.
- or select the control, then go to the property grid Misc-> Name and rename it there.

**Reply by UniaxialMotion** [reply, marked as answer]
To add onto this: I was struggling with this as well, and had my root element already named when I had used right-click > Create Library from selection, but my library still didn't appear in the editor. Renaming the root element again (to something else and back) seems to have resolved the issue for me.
